src/main/java/org/junit/rules/TemporaryFolder.java

```diff
@@ -216,10 +219,23 @@ public File newFolder() throws IOException {
     }
 
     private File createTemporaryFolderIn(File parentFolder) throws IOException {
-        File createdFolder = File.createTempFile("junit", "", parentFolder);
-        createdFolder.delete();
-        createdFolder.mkdir();
-        return createdFolder;
+        File createdFolder = null;
+        for (int i = 0; i < TEMP_DIR_ATTEMPTS; ++i) {
+            // Use createTempFile to get a suitable folder name.
+            final String suffix = ".tmp";
+            final File tmpFile = File.createTempFile(TMP_PREFIX, suffix, parentFolder);
+            tmpFile.delete();
```

Comment:
In my proposal, the temporary file was deleted after the temporary directory was created. That way we wouldn't have a recent condition if the user is using an older JDK where `File.createTempFile()` was sometimes returning paths that were returned from previous calls.